src/Cardano/Wallet/BlockSyncer.hs

```diff
@@ -0,0 +1,158 @@
+                killThread threadID
+            else do
+                threadDelay (period*1000*1000)
+                writeIORef counter $ min (length blocks - 1) (previousCounter + 1)
+                countDown counter
+
+
+_getBlock
+    :: [((Hash "BlockHeader"),Block)]
+    -> (Hash "BlockHeader")
+    -> IO Block
+_getBlock blocks headerHash =
+    case filter (\(h,_) -> h == headerHash) blocks of
+        [(_,block)] -> return block
+        _ -> error "Something wrong with the test blocks"
+
+_getEpoch
+    :: [((Hash "BlockHeader"),Block)]
+    -> Word64
+    -> IO [Block]
+_getEpoch blocks epoch = do
+    let bs =  filter (\(_,Block (BlockHeader theEpoch _ _) _) -> theEpoch == epoch) blocks
+    pure $ map snd bs
+
+_getNetworkTip
+    :: [((Hash "BlockHeader"),Block)]
+    -> IORef Int
+    -> IO ((Hash "BlockHeader"), BlockHeader)
+_getNetworkTip blocks currentPosition = do
+    index <- readIORef currentPosition
+    let (headerHash, Block blockHeader _ ) = (reverse blocks) !! index
+    pure (headerHash,blockHeader)
+
+blockSyncerBracket
+    :: IORef BlocksConsumed
+    -> BlockLayerMockup
+    -> Int
+    -> IO ()
+blockSyncerBracket blocksPersisted blockLayer period =
+    bracket
+      (return ()) --(putStrLn "setting up for ticking function")
+      (\_->return ()) --(\_ -> putStrLn "cleaning for ticking function")
+      (tickingFunction blocksPersisted blockLayer period)
+
+tickingFunction
+    :: IORef BlocksConsumed
+    -> BlockLayerMockup
+    -> Int
+    -> ()
+    -> IO ()
+tickingFunction initialState blockLayer delayPeriod _ =
+    go initialState
+    `catch`
+    (\(e :: SomeException) ->
+            let msg = "Terminating tickingFunction of BlockSyncer due to " % F.shown
+            in (putStrLn . showText) $ sformat msg e
+    )
+    where
+      go
+          :: IORef BlocksConsumed
+          -> IO ()
+      go previousState  = do
+          --putStrLn "syncing the block ..."
+          (BlocksConsumed headerHashesConsumed) <- readIORef initialState
+          (currentHeaderHash, BlockHeader _ _ previousHeaderHash) <- getNetworkTip blockLayer
+          updatedHeaderHashes <-
+              case headerHashesConsumed of
+                  [] -> pure [currentHeaderHash]
+                  lastConsumed : _ | lastConsumed == currentHeaderHash ->
+                                         pure headerHashesConsumed
+                                   | lastConsumed == previousHeaderHash ->
+                                         pure $ currentHeaderHash : headerHashesConsumed
+                                   | otherwise -> do
+                                         missingHashes <- retrieveMissingBlocks [previousHeaderHash]
+                                                          headerHashesConsumed
+                                                          blockLayer
+                                         pure $ currentHeaderHash : (missingHashes ++ headerHashesConsumed)
```

Comment:
Make sure to follow the coding standards on variable-length indentation :pray:

Reply:
:+1: